packages/react/src/template.tsx

```diff
@@ -15,23 +15,17 @@ export type ComponentDataState = {
   $: 'component'
   key?: string
   component: ComponentIdentifier
-  children?: JSONValue
-  props?: Record<string, JSONValue>
+  children?: DataState
+  props?: Record<string, DataState>
 }
 type ReferencedComponentDataState = ComponentDataState & {
-  ref: ReferencedDataState['ref']
+  path: Path
 }
+export type Path = string | [string, ...(string | number)[]]
```

Comment:
Yeah, this is because Path is used to reference data in a store and can be either string, e.g. ["myStore", "field"] or a number, when accessing arrays, e.g. ["myStore", "list", 0, "title"]. 

However, by design, the first value MUST be a string. That is because store names are currently typed as strings.

Reply:
Ok makes sense

Reply:
Also, just a note, typing as `[string, ...(string[])]` looks odd, but makes TypeScript happy by ensuring you can do `path[0]` and that value is always a string.